byterun/shared_heap.c

```diff
@@ -188,25 +186,39 @@ static void pool_release(struct caml_heap_state* local, pool* pool, sizeclass sz
   Assert(pool->sz == sz);
   local->stats.pool_words -= POOL_WSIZE;
   local->stats.pool_frag_words -= POOL_HEADER_WSIZE + wastage_sizeclass[sz];
-  if (local->num_free_pools < MAX_LOCAL_FREE_POOLS) {
-    /* FIXME: in the local cache, other domains can't get it */
-    local->num_free_pools++;
-    pool->next = local->free_pools;
-    local->free_pools = pool;
-  } else {
-    /* FIXME: give free pools back to the OS */
-    caml_plat_lock(&pool_freelist.lock);
-    pool->next = pool_freelist.free;
-    pool_freelist.free = pool;
-    caml_plat_unlock(&pool_freelist.lock);
+  /* FIXME: give free pools back to the OS */
```

Comment:
Given that local->free_pools and local->num_free_pools are being factored out in this change. To remove future confusion, can we also remove them from the code in pool_acquire?

Reply:
Yes, I have now removed `free_pools` and `num_free_pools` entirely.